### DISCUSSION TITLE
[BE] 고객의 방문횟수를 계산 관련

### DISCUSSION THREAD

**Discussion by yenawee** [woowacourse-teams/2023-stamp-crush#173] [General]
현재 ERD 를 기반으로 고객의 방문횟수를 계산하기에 로직이 너무 복잡해짐.

visited 테이블을 새로 생성할 지 논의가 필요

**Comment by gitchannn** [comment]
**깃짱의 의견**

- 방문수는 사장님 입장에서 꽤 중요한 지표 중 하나일 것이라서 반드시 제공해야 한다고 생각함.
- 현재 로직은 스탬프가 입력된 시간을 기준으로 방문수를 계산하는 방식인데, 중요한 지표를 계산하는데 있어서 정확도도 떨어지고, 불필요하게 많은 로직이 들어가서 비효율적이라고 생각함.
- **하나의 테이블을 더 추가해서 만드는 게 좋다고 생각함.**

**Comment by youngh0** [comment]
현재 두 가지 로직이 모두 구현되어 있지 않은 상태에서 분 단위로 오차 계산해서 짜는 것보단 새로운 테이블을 추가하는게 덜 힘들고, 보다 정확한 방문 횟수 데이터를 제공할 수 있다고 생각합니다.

시간 오차범위로 구현하는 것은 허점이 있다고 생각합니다.

**Comment by gitchannn** [comment]
하디: 쿠폰에 방문 횟수를 넣는건 어떤가요?

깃짱:
1. 쿠폰이 변경되는 경우에 이전 쿠폰에 입력된 방문횟수를 가져와야 한다는 점에서 또 비효율이 발생할 수 있음
2. 쿠폰이랑 방문횟수는 좀 주제(?)가 다른 데이터인 것 같아서 한 테이블에 묶는게 직관적이지 않게 느껴짐

**Comment by gitchannn** [comment]
# ERD를 바꾸자

**Comment by gitchannn** [comment]
**VISIT_COUNT**
cafe_id
customer_id

**Reply by gitchannn** [reply]
cafe_id 1 
customer_id 1 
visited_at (created_at과 같은 내용)

이런식으로 삭제 없이 데이터를 쌓기만 하고,
나중에 방문일자를 구할 때는 cafe_id, customer_id로 필터링해서 count

**Reply by gitchannn** [reply]
_visited_at은 의미상이고, 실제 테이블에는 created_at으로 사용하기로 결정_

**Comment by gitchannn** [comment]
# 테이블 이름 투표

**Reply by gitchannn** [reply]
VISIT_HISTORY

**Reply by gitchannn** [reply]
VISIT_DATA
